src/nbl/video/CVulkanRayTracingPipeline.cpp

```diff
@@ -0,0 +1,52 @@
+#include "nbl/video/CVulkanRayTracingPipeline.h"
+
+#include "nbl/video/CVulkanLogicalDevice.h"
+
+namespace nbl::video
+{
+
+  CVulkanRayTracingPipeline::CVulkanRayTracingPipeline(
+    const SCreationParams& params, 
+    const VkPipeline vk_pipeline, 
+    ShaderGroupHandleContainer&& shaderGroupHandles) :
+    IGPURayTracingPipeline(params),
+    m_vkPipeline(vk_pipeline),
+    m_shaders(core::make_refctd_dynamic_array<ShaderContainer>(params.shaders.size())),
+    m_shaderGroupHandles(std::move(shaderGroupHandles))
+  {
+    for (size_t shaderIx = 0; shaderIx < params.shaders.size(); shaderIx++)
+      m_shaders->operator[](shaderIx) = ShaderRef(static_cast<const CVulkanShader*>(params.shaders[shaderIx].shader));
+
```

Comment:
please query the stack size for every shader in every group with 
https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#vkGetRayTracingShaderGroupStackSizeKHR

Reply:
Important caveat, you cannot call `vkGetRayTracingShaderGroupStackSizeKHR` on shaders which are `VK_SHADER_UNUSED_KHR`
https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#VUID-vkGetRayTracingShaderGroupStackSizeKHR-groupShader-03609

so check before you call it, if unused the stack size must be 0 I guess